core/src/main/java/hudson/model/ViewGroupMixIn.java

```diff
@@ -99,6 +101,15 @@ public View getView(String name) {
             View pv = getPrimaryView();
             if (pv instanceof ViewGroup)
                 return ((ViewGroup)pv).getView(name);
+            if (pv instanceof AllView && AllView.DEFAULT_VIEW_NAME.equals(pv.name)) {
```

Comment:
Sounds right but hard for me at least to follow how this will work in progress; recommend having some code coverage using `@LocalData` to prove it works.